ansys/mapdl/core/logging.py

```diff
@@ -0,0 +1,499 @@
+        ----------
+        filename : str, optional
+            Name of the file where the logs are recorded. By default FILE_NAME
+        level : str, optional
+            Level of logging. E.x. 'DEBUG'. By default LOG_LEVEL
+        """
+
+        self = add_file_handler(self, filename=filename, level=level, write_headers=True)
+
+    def log_to_stdout(self, level=LOG_LEVEL):
+        """
+        Add standard output handler to the logger.
+
+        Parameters
+        ----------
+        level : str, optional
+            Level of logging record. By default LOG_LEVEL
+        """
+
+        self = add_stdout_handler(self, level=level)
+
+    ## Copy functions
+    add_stdout_handler = log_to_stdout
+    add_file_handler = log_to_file
+
+    def _make_child_logger(self, sufix, level):
+        """Make a child logger, either using ``getChild`` or copying attributes between ``_global_``
+        logger and the new one. """
+        logger = logging.getLogger(sufix)
+        logger._std_out_handler = None
+        logger._file_handler = None
+
+        if self._global.hasHandlers:
+            for each_handler in self._global.handlers:
+                new_handler = copy(each_handler)
+
+                if each_handler == self._file_handler:
+                    logger._file_handler = new_handler
+                elif each_handler == self._std_out_handler:
+                    logger._std_out_handler = new_handler
+
+                if level:
+                    # The logger handlers are copied and changed the loglevel is the specified log level
+                    # is lower than the one of the global.
+                    if each_handler.level > string_to_loglevel[level.upper()]:
+                        new_handler.setLevel(level)
+
+                logger.addHandler(new_handler)
+
+        if level:
+            if isinstance(level, str):
+                level = string_to_loglevel[level.upper()]
+            logger.setLevel(level)
+
+        else:
+            logger.setLevel(self._global.level)
+
+        logger.propagate = True
+        return logger
+
+    def add_child_logger(self, sufix, level=None):
+        """
+        Add a child logger to the main logger. This logger is more general than
+        an instance logger which is designed to track the state of the MAPDL instances.
+
+        If the logging level is in the arguments, a new logger with a reference to the ``_global`` logger handlers
+        is created instead of a child.
+
+        Parameters
+        ----------
+        sufix : str
+            Name of the logger.
+
+               level : str
+            Level of logging
+
+        Returns
+        -------
+        logging.logger
+            Logger class.
+        """
+        name = self._global.name + '.' + sufix
+        self._instances[name] = self._make_child_logger(self, name, level)
+        return self._instances[name]
+
+    def _add_MAPDL_instance_logger(self, name, MAPDL_instance, level):
```

Comment:
Parameters should always be lowercase snakecase.  Same with methods and functions.

Reply:
Fixed.